### PR TITLE
Upgrade python dependencies

### LINKED ISSUE #1469: Upgrade Python dependencies to fix vulnerabilities
# Description

Upgrade any Python dependencies that have known vulnerabilities.

## Tasks

- [x] Run snyk and trivy, fix as many of vulns as we can (0d) - @shreyamalviya 
- [x] Test the monkey with updated dependencies and verify it still runs on ubuntu 16 (0d) @ilija-lazoroski 
- [x] Open issues with suggestions for the security bugs we were unable to fix (timeboxed 0d) @ilija-lazoroski, @mssalvatore 
  - [x] Do a research on how to implement [different virtual environments ](https://stackoverflow.com/questions/48332046/does-pip-python-support-multiple-versions-of-the-same-package) and [environments markers ](https://www.python.org/dev/peps/pep-0508/#environment-markers) to help with different version of python packages for 32bit and 64bit agent.

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

Fixes #1469

All vulnerabilities in `infection_monkey/` have been resolved. The packages in `monkey_island/` having vulnerabilities can't be updated since `awscli`'s version is fixed to 1.18.131, which needs specific versions of other packages. Why `awscli`'s version was fixed and whether we can update it still needs to be investigated.

`monkey_island/` vulnerabilities:
image

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/infection_monkey/Pipfile [hunk 1]
```diff
@@ -5,7 +5,6 @@ name = "pypi"
 
 [packages]
 altgraph = "*"
-cryptography = "==2.5" # We can't build 32bit ubuntu12 binary with newer versions of cryptography
```
Comment (mssalvatore) [inline monkey/infection_monkey/Pipfile:8]: @shreyamalviya Have you tried actually building the monkey binaries with this change?

Comment (ilija-lazoroski) [reply monkey/infection_monkey/Pipfile:8]: I will do the tests now.

Comment (shreyamalviya) [reply monkey/infection_monkey/Pipfile:8]: No, that's the next task in the issue. I'll do it tomorrow if no one else does it by then.

Comment (ilija-lazoroski) [reply monkey/infection_monkey/Pipfile:8]: Linux 32bit failed...

**Comment by ilija-lazoroski** [comment]
New PR coming.

**Pull request closed without merging**